driver-sync/src/test/functional/com/mongodb/client/unified/Entities.java

```diff
@@ -128,135 +217,36 @@ public TestCommandListener getClientCommandListener(final String id) {
         return entity;
     }
 
+    private <T> void putEntity(final String id, final T entity, final Map<String, T> entities) {
+        if (!entityNames.add(id)) {
+            throw new IllegalStateException("Already an entity with id: " + id);
+        }
+        entities.put(id, entity);
+    }
+
     public void init(final BsonArray entitiesArray, final Function<MongoClientSettings, MongoClient> mongoClientSupplier) {
         for (BsonValue cur : entitiesArray.getValues()) {
             String entityType = cur.asDocument().getFirstKey();
             BsonDocument entity = cur.asDocument().getDocument(entityType);
             String id = entity.getString("id").getValue();
             switch (entityType) {
                 case "client":
-                    MongoClientSettings.Builder clientSettingsBuilder = getMongoClientSettingsBuilder();
```

Comment:
I had to refactor this method because it got too long.  So long that even Checkstyle complained.